api/pkg/s3/objectacl.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"encoding/xml"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/emicklei/go-restful"
+	"github.com/journeymidnight/yig/helper"
+	"github.com/opensds/multi-cloud/api/pkg/common"
+	"github.com/opensds/multi-cloud/api/pkg/s3/datatype"
+	s3error "github.com/opensds/multi-cloud/s3/error"
+	pb "github.com/opensds/multi-cloud/s3/proto"
+	log "github.com/sirupsen/logrus"
+)
+
+func (s *APIService) ObjectAclPut(request *restful.Request, response *restful.Response) {
+	bucketName := strings.ToLower(request.PathParameter(common.REQUEST_PATH_BUCKET_NAME))
+	objectKey := request.PathParameter(common.REQUEST_PATH_OBJECT_KEY)
+	log.Infof("received request: PUT bucket[name=%s] object[name=%s] acl\n", bucketName, objectKey)
+
+	var err error
+	var acl datatype.Acl
+	var policy datatype.AccessControlPolicy
+	if _, ok := request.Request.Header[common.REQUEST_HEADER_ACL]; ok {
+		acl, err = getAclFromHeader(request)
+		if err != nil {
+			WriteErrorResponse(response, request, err)
+			return
+		}
+	} else {
+		aclBuffer, err := ioutil.ReadAll(io.LimitReader(request.Request.Body, 1024))
```

Comment:
Add comment to explain why use 1024 here please.

Reply:
thanks for your reviewing,  I will do it